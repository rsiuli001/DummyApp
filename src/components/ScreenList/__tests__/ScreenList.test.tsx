import { shallow } from 'enzyme';
import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import { ScreenList } from '../../';

describe('Screen List', () => {
  describe('rendering list', () => {
    const onPressEvent = jest.fn();
    const wrapper = shallow(
      <ScreenList
        text="screenName"
        color={'red'}
        onPress={onPressEvent}
        screenName="screenName"
      />
    );

    it('should render the container view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(View).length).toEqual(1);
      expect(wrapper.find(TouchableOpacity).length).toEqual(1);
    });
  });

  describe('Interaction', () => {
    it('onPress event works', () => {
      const onPressEvent = jest.fn();
      const wrapper = shallow(
        <ScreenList
          text="screenName"
          color={'red'}
          onPress={onPressEvent}
          screenName="screenName"
        />
      );

      wrapper.find(TouchableOpacity).first().prop('onPress')!({} as any);
      expect(onPressEvent.mock.calls.length).toBe(1);
    });
  });
});
