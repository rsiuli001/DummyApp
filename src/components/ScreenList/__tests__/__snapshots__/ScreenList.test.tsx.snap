// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen List rendering list should render the container view 1`] = `ShallowWrapper {}`;
